tests/utility_scripts/test_utility_scripts.py

```diff
@@ -66,3 +66,73 @@ def test_pick(position, n_elements, stack, expected):
 
     assert context.evaluate()
     assert len(context.get_altstack()) == 0
+
+
+@pytest.mark.parametrize(
+    ("is_from_alt", "is_mod_on_top", "is_constant_reused", "is_positive", "stack", "expected"),
+    [
+        (False, False, False, False, [5, -7], [-2]),
+        (False, False, False, True, [5, -7], [3]),
+        (False, False, True, False, [5, -7], [5, -2]),
+        (False, False, True, True, [5, -7], [5, 3]),
+        (False, True, False, False, [-7, 5], [-2]),
+        (False, True, False, True, [-7, 5], [3]),
+        (False, True, True, False, [-7, 5], [5, -2]),
+        (False, True, True, True, [-7, 5], [5, 3]),
+        (True, False, False, False, [-7, 3, 5], [3, -2]),
+        (True, False, False, True, [-7, 3, 5], [3, 3]),
+        (True, False, True, False, [-7, 3, 5], [3, 5, -2]),
+        (True, False, True, True, [-7, 3, 5], [3, 5, 3]),
+        (True, True, False, False, [5, 3, -7], [3, -2]),
+        (True, True, False, True, [5, 3, -7], [3, 3]),
+        (True, True, True, False, [5, 3, -7], [3, 5, -2]),
+        (True, True, True, True, [5, 3, -7], [3, 5, 3]),
+        (False, False, False, False, [13, -17], [-4]),
+        (False, False, False, True, [13, -17], [9]),
+        (False, False, True, False, [13, -17], [13, -4]),
+        (False, False, True, True, [13, -17], [13, 9]),
+        (False, True, False, False, [-17, 13], [-4]),
+        (False, True, False, True, [-17, 13], [9]),
+        (False, True, True, False, [-17, 13], [13, -4]),
+        (False, True, True, True, [-17, 13], [13, 9]),
+    ],
+)
```

Comment:
Can't we have a altstack in the parameters?

Reply:
done